usr.sbin/bsdinstall/scripts/bootconfig

```diff
@@ -117,21 +117,29 @@ if [ -n "$(awk '{if ($2=="/boot/efi") printf("%s\n",$1);}' $PATH_FSTAB)" ]; then
 
 	f_dprintf "Installing loader.efi onto ESP"
 	mkdir -p "${mntpt}/${FREEBSD_BOOTDIR}" "${mntpt}/${BOOTDIR}"
-	cp "$BSDINSTALL_CHROOT/boot/loader.efi" "${mntpt}/${FREEBSD_BOOTNAME}"
+
+	if [ -f "$BSDINSTALL_CHROOT/boot/loader_ia32.efi" ] && [ `sysctl -n machdep.efi_arch` == i386 ]; then
```

Comment:
Why not hoist this up so you can just set the variables to the right thing in the first place?